google-cloud-spanner/lib/google/cloud/spanner/database/job.rb

```diff
@@ -122,6 +122,32 @@ def error?
           end
 
           ##
+          # The status code and message if the operation associated with this
+          # job produced an error, returned as an error object.
+          #
+          # @return [Google::Cloud::Error, nil] An error object containing the
+          #   status code and message, or `nil` if no error occurred.
+          #
+          # @example
+          #   require "google/cloud/spanner"
+          #
+          #   spanner = Google::Cloud::Spanner.new
+          #
+          #   job = spanner.create_database "my-instance",
+          #                                 "my-new-database"
+          #
+          #   job.error? # true
+          #
+          #   error = job.error
+          #
+          def error
+            return nil unless error?
+            fail GRPC::BadStatus.new @grpc.error.code, @grpc.error.message
```

Comment:
In order to set `Error#cause` we need to raise the error. So I suppose this code should look more like this:

```ruby
begin
  begin
    fail GRPC::BadStatus.new @grpc.error.code, @grpc.error.message
  rescue GRPC::BadStatus => grpc_error
    fail Google::Cloud::Error.from_error grpc_error
  end
rescue Google::Cloud::Error => error
  return error
end
```

Or, we could just return the Google::Rpc::Status object, leaking the low level API.